web/src/browser_history.js

```diff
@@ -96,3 +98,29 @@ export function update_hash_internally_if_required(hash) {
 export function return_to_web_public_hash() {
     window.location.hash = state.spectator_old_hash;
 }
+
+export const hash_to_page_location = {};
+
+export function save_narrow() {
+    const narrow_data = {};
+    const $row = message_lists.home.selected_row();
```

Comment:
I think this shouldn't be outside the if/else?

Reply:
Yup, good catch!